spec/models/provider_spec.rb

```diff
@@ -34,38 +34,146 @@
   end
 
   describe '#address_info' do
-    it 'returns address of the provider' do
-      provider = create(:provider, enrichments: [])
-      expect(provider.address_info).to eq(
-        'address1' => provider.address1,
-        'address2' => provider.address2,
-        'address3' => provider.address3,
-        'address4' => provider.address4,
-        'postcode' => provider.postcode,
-        'region_code' => provider.region_code
-      )
-    end
-
-    context 'provider has enrichments' do
-      it 'returns json_data from the last enrichment' do
-        enrichment = build(:provider_enrichment)
-        provider = create(:provider, enrichments: [enrichment])
+    context 'empty enrichments' do
+      it 'returns address of the provider' do
+        provider = create(:provider, enrichments: [])
 
         expect(provider.address_info).to eq(
-          'address1' => enrichment.address1,
-          'address2' => enrichment.address2,
-          'address3' => enrichment.address3,
-          'address4' => enrichment.address4,
-          'postcode' => enrichment.postcode,
-          'region_code' => enrichment.region_code
+          'address1' => provider.address1,
+          'address2' => provider.address2,
+          'address3' => provider.address3,
+          'address4' => provider.address4,
+          'postcode' => provider.postcode,
+          'region_code' => ProviderEnrichment.region_codes[provider.region_code]
         )
       end
+
+      context 'provider has enrichments' do
+        context 'enrichment has nothing set for address_info' do
+          it 'returns address of the provider' do
+            enrichment = build(:provider_enrichment, json_data: { 'email' => Faker::Internet.email,
+            'website' => Faker::Internet.url,
+            'train_with_us' => Faker::Lorem.sentence.to_s,
+            'train_with_disability' => Faker::Lorem.sentence.to_s })
+            provider = create(:provider, enrichments: [enrichment])
+
+            # forcing it to be absent in json_data altogether
+            ProviderEnrichment.connection.update(<<~EOSQL)
+              UPDATE provider_enrichment
+                    SET json_data=json_data-'Address1'-'Address2'-'Address3'-'Address4'-'Postcode'-'RegionCode'
+                    WHERE provider_code='#{enrichment.id}'
+            EOSQL
+
+            expect(provider.address_info).to eq(
+              'address1' => provider.address1,
+              'address2' => provider.address2,
+              'address3' => provider.address3,
+              'address4' => provider.address4,
+              'postcode' => provider.postcode,
+              'region_code' => ProviderEnrichment.region_codes[provider.region_code]
+            )
+          end
+          it 'returns address of the provider' do
+            enrichment = build(:provider_enrichment, json_data: { 'email' => Faker::Internet.email,
```

Comment:
Set the attributes you want here, `address1`, `address2`, etc as object attributes, not as `json_data` attributes;